### PR TITLE
Add changelog for 0.4

### PR DESCRIPTION
**Pull request by xing-yang**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
- Add changelog for 0.4.0
- Update image version used in deployment yaml to v0.4.0

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-0.4.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+# Release notes for v0.4.0
+
+[Documentation](https://docs.k8s.io/docs/home)
```
Comment (chrishenzie) [inline CHANGELOG/CHANGELOG-0.4.md:3]: I think documentation should point at https://kubernetes-csi.github.io/

Comment (xing-yang) [reply CHANGELOG/CHANGELOG-0.4.md]: Fixed.

**Comment by xing-yang** [comment]
@chrishenzie addressed your comments.  PTAL.

**Pull request merged**